coderd/autobuild/lifecycle_executor.go

```diff
@@ -142,7 +142,7 @@ func (e *Executor) runOnce(t time.Time) Stats {
 	// NOTE: If a workspace build is created with a given TTL and then the user either
 	//       changes or unsets the TTL, the deadline for the workspace build will not
 	//       have changed. This behavior is as expected per #2229.
-	workspaces, err := e.db.GetWorkspacesEligibleForTransition(e.ctx, t)
+	workspaces, err := e.db.GetWorkspacesEligibleForTransition(e.ctx, currentTick)
```

Comment:
In our lifecycle executor we use `currentTick` for checking eligibility, so it makes sense to also use it for our database query.